release-content/0.15/release-notes/function_reflection.md

```diff
@@ -0,0 +1,214 @@
+    FunctionInfo::named("add_to")
+        .with_arg::<i32>("a")
+        .with_arg::<&mut i32>("b")
+        .with_return::<()>(),
+);
+```
+
+#### The Function Registry
+
+To make it easier to work with reflected functions, a dedicated [`FunctionRegistry`] has been added.
+This works similarly to the [`TypeRegistry`] where functions can be registered and retrieved by name.
+
+```rust
+let mut registry = FunctionRegistry::default();
+registry
+    // Named functions can be registered directly
+    .register(add)?
+    // Unnamed functions (e.g. closures) must be registered with a name
+    .register_with_name("add_3", |a: i32, b: i32, c: i32| a + b + c)?;
+
+let add = registry.get("my_crate::math::add").unwrap();
+let add_3 = registry.get("add_3").unwrap();
+```
+
+For better integration with the rest of Bevy, a new [`AppFunctionRegistry`] resource has been added
+along with registration methods on [`App`].
+
+#### The `Function` Trait
+
+A new reflection trait—appropriately called [`Function`]—has been added to correspond to functions.
+
+Due to limitations in Rust, we're unable to implement this trait for all functions,
+but it does make it possible to pass around a `DynamicFunction` as a [`PartialReflect`] trait object.
+
+```rust
+#[derive(Reflect)]
+#[reflect(from_reflect = false)]
+struct EventHandler {
+    callback: DynamicFunction<'static>,
+}
+
+let event_handler: Box<dyn Struct> = Box::new(EventHandler {
+    callback: (|| println!("Event fired!")).into_function(),
+});
+
+let field = event_handler.field("callback").unwrap();
+
+if let ReflectRef::Function(callback) = field.reflect_ref() {
+    callback.reflect_call(ArgList::new()).unwrap();
+}
+```
+
+#### Limitations
+
+While this feature is quite powerful already, there are still a number of limitations.
+
+Firstly, `IntoFunction`/`IntoFunctionMut` only work for functions with up to 16 arguments,
+and only support returning borrowed data where the lifetime is tied to the first argument
+(normally `self` in methods).
+
+Secondly, the `Function` trait can't be implemented for all functions due to how the function
+reflection traits are defined.
+
+Thirdly, all arguments and return types must have derived `Reflect`.
+This can be confusing for certain types such as `&str` since only `&'static str` implements
+`Reflect` and its borrowed version would be `&&'static str`.
+
+Lastly, while generic functions are supported, they must first be manually monomorphized.
+This means that if you have a generic function like `fn foo<T>()`, you have to create the
+`DynamicFunction` like `foo::<i32>.into_function()`.
+
+Most of these limitations are due to Rust itself.
+The [lack of variadics] and [issues with coherence] are among the two biggest difficulties
+to work around.
+Despite this, we will be looking into ways of improving the ergonomics and capabilities 
+of this feature in future releases.
+
+We already have a PR up to add support for overloaded functions: functions with a variable
```

Comment:
Link here!

Reply:
```suggestion
We already have a [PR](https://github.com/bevyengine/bevy/pull/15074) up to add support for overloaded functions: functions with a variable
```

Note: Using an inline link to avoid any possible issues with `PR` across the rest of the post (I'm not sure how links are generated, but just to be safe)

Reply:
They'll only operate based on [TEXT] tags with square brackets, but this is fine.